src/components/HTMLEngineProvider/HTMLRenderers/MentionReportRenderer.tsx

```diff
@@ -0,0 +1,116 @@
+import isEmpty from 'lodash/isEmpty';
+import React from 'react';
+import type {TextStyle} from 'react-native';
+import {StyleSheet} from 'react-native';
+import type {OnyxCollection, OnyxEntry} from 'react-native-onyx';
+import {withOnyx} from 'react-native-onyx';
+import type {CustomRendererProps, TPhrasing, TText} from 'react-native-render-html';
+import {ShowContextMenuContext} from '@components/ShowContextMenuContext';
+import Text from '@components/Text';
+import useCurrentReportID from '@hooks/useCurrentReportID';
+import useStyleUtils from '@hooks/useStyleUtils';
+import useThemeStyles from '@hooks/useThemeStyles';
+import Navigation from '@navigation/Navigation';
+import CONST from '@src/CONST';
+import ONYXKEYS from '@src/ONYXKEYS';
+import ROUTES from '@src/ROUTES';
+import type {Report} from '@src/types/onyx';
+import {isEmptyObject} from '@src/types/utils/EmptyObject';
+
+type RoomMentionOnyxProps = {
+    /** All reports shared with the user */
+    reports: OnyxCollection<Report>;
+};
+
+type MentionRoomRendererProps = RoomMentionOnyxProps & CustomRendererProps<TText | TPhrasing>;
+
+const removeLeadingLTRAndHash = (value: string) => value.replace(CONST.UNICODE.LTR, '').slice(1);
+
+const getMentionDetails = (htmlAttributeReportID: string, currentReportID: string, reports: OnyxCollection<Report>, tnode: TText | TPhrasing) => {
+    let reportID: string | undefined;
+    let mentionDisplayText: string;
+
+    // get mention details based on reportID from tag attribute
+    if (!isEmpty(htmlAttributeReportID)) {
+        const report = reports?.['report_'.concat(htmlAttributeReportID)];
+
+        reportID = report?.reportID ?? undefined;
+        mentionDisplayText = report?.reportName ?? report?.displayName ?? htmlAttributeReportID;
+        // get mention details from name inside tnode
+    } else if ('data' in tnode && !isEmptyObject(tnode.data)) {
+        mentionDisplayText = removeLeadingLTRAndHash(tnode.data);
+
+        const currentReport = reports?.['report_'.concat(currentReportID)];
+        // eslint-disable-next-line rulesdir/prefer-early-return
+        Object.values(reports ?? {}).forEach((report) => {
+            if (report?.policyID === currentReport?.policyID && removeLeadingLTRAndHash(report?.reportName ?? '') === mentionDisplayText) {
+                reportID = report?.reportID;
+            }
+        });
+    } else {
+        return null;
+    }
+
+    return {reportID, mentionDisplayText};
+};
+
+function MentionReportRenderer({style, tnode, TDefaultRenderer, reports, ...defaultRendererProps}: MentionRoomRendererProps) {
+    const styles = useThemeStyles();
+    const StyleUtils = useStyleUtils();
+    const htmlAttributeReportID = tnode.attributes.reportid;
+    const currentReportID = useCurrentReportID();
```

Comment:
You can destructure here
```suggestion
    const {currentReportID} = useCurrentReportID();
```